cardano-config/src/Cardano/Config/ShelleyGenesis.hs

```diff
@@ -0,0 +1,243 @@
+    }
+  where
+    k = 2160
+    zeroTime = Time.UTCTime (Time.fromGregorian 1970 1 1) 0 -- tradition
+
+
+--
+-- ShelleyGenesis JSON conversion
+--
+
+shelleyGenesisToJSON :: ShelleyGenesis TPraosStandardCrypto -> Value
+shelleyGenesisToJSON = toJSON
+
+shelleyGenesisFromJSON :: Value -> Parser (ShelleyGenesis TPraosStandardCrypto)
+shelleyGenesisFromJSON = parseJSON
+
+instance Crypto crypto => ToJSON (ShelleyGenesis crypto) where
+  toJSON sg =
+    Aeson.object
+      [ "StartTime"             .= sgStartTime sg
+        --TODO: this should not have both network magic and protocol magic
+        -- they are different names for the same thing used in two ways.
+      , "NetworkMagic"          .= sgNetworkMagic sg
+      , "ProtocolMagicId"       .= sgProtocolMagicId sg
+      , "ActiveSlotsCoeff"      .= sgActiveSlotsCoeff sg
+      , "DecentralisationParam" .= sgDecentralisationParam sg
+      , "SecurityParam"         .= sgSecurityParam sg
+      , "EpochLength"           .= sgEpochLength sg
+      , "SlotsPerKESPeriod"     .= sgSlotsPerKESPeriod sg
+      , "MaxKESEvolutions"      .= sgMaxKESEvolutions sg
+      , "SlotLength"            .= sgSlotLength sg
+      , "UpdateQuorum"          .= sgUpdateQuorum sg
+      , "MaxMajorPV"            .= sgMaxMajorPV sg
+      , "MaxLovelaceSupply"     .= sgMaxLovelaceSupply sg
+      , "MaxBodySize"           .= sgMaxBodySize sg
+      , "MaxHeaderSize"         .= sgMaxHeaderSize sg
+      , "GenDelegs"             .= sgGenDelegs sg
+      , "InitialFunds"          .= sgInitialFunds sg
+      ]
+
+instance Crypto crypto => FromJSON (ShelleyGenesis crypto) where
+  parseJSON =
+    Aeson.withObject "ShelleyGenesis" $ \ obj ->
+      ShelleyGenesis
+        <$> obj .: "StartTime"
+        <*> obj .: "NetworkMagic"
+        <*> obj .: "ProtocolMagicId"
+        <*> obj .: "ActiveSlotsCoeff"
+        <*> obj .: "DecentralisationParam"
+        <*> obj .: "SecurityParam"
+        <*> obj .: "EpochLength"
+        <*> obj .: "SlotsPerKESPeriod"
+        <*> obj .: "MaxKESEvolutions"
+        <*> obj .: "SlotLength"
+        <*> obj .: "UpdateQuorum"
+        <*> obj .: "MaxMajorPV"
+        <*> obj .: "MaxLovelaceSupply"
+        <*> obj .: "MaxBodySize"
+        <*> obj .: "MaxHeaderSize"
+        <*> obj .: "GenDelegs"
+        <*> obj .: "InitialFunds"
+
+
+--
+-- Simple newtype wrappers JSON conversion
+--
+
+deriving newtype instance Aeson.ToJSON   Coin
+deriving newtype instance Aeson.FromJSON Coin
+
+deriving newtype instance Aeson.ToJSON   EpochSize
+deriving newtype instance Aeson.FromJSON EpochSize
+
+deriving newtype instance Aeson.ToJSON   NetworkMagic
+deriving newtype instance Aeson.FromJSON NetworkMagic
+
+deriving newtype instance Aeson.ToJSON   SecurityParam
+deriving newtype instance Aeson.FromJSON SecurityParam
+
+deriving newtype instance Aeson.ToJSON   SlotLength
+deriving newtype instance Aeson.FromJSON SlotLength
+
+deriving newtype instance Aeson.ToJSON   SystemStart
```

Comment:
I would prefer explicit instances as they make life easier when debugging things. Although the types appear simple enough to just automatically derive them. @erikd thoughts?